ray-operator/controllers/ray/utils/util.go

```diff
@@ -71,6 +71,38 @@ func IsCreated(pod *corev1.Pod) bool {
 	return pod.Status.Phase != ""
 }
 
+func FindPodReadyCondition(pod *corev1.Pod, condType rayv1.RayClusterConditionType) metav1.Condition {
+	replicaPodReadyCondition := metav1.Condition{
+		Type:   string(condType),
+		Status: metav1.ConditionFalse,
+		Reason: rayv1.UnknownReason,
+	}
+
+	for _, cond := range pod.Status.Conditions {
+		if cond.Type != corev1.PodReady {
+			continue
+		}
+		// Set the status based on the PodReady condition
+		replicaPodReadyCondition.Status = metav1.ConditionStatus(cond.Status)
+		replicaPodReadyCondition.Message = cond.Message
+
+		// Determine the reason; default to PodRunningAndReady if the pod is ready but no specific reason is provided
+		reason := cond.Reason
+		if cond.Status == corev1.ConditionTrue && reason == "" {
+			reason = rayv1.PodRunningAndReady
+		}
+
+		// Update the reason if it's not empty
+		if reason != "" {
```

Comment:
this seems to be unnecessary?

Reply:
it could happen when `cond.Status != corev1.ConditionTrue` and `cond.Reason` give us empty which i think is pretty rare when we are checking `cond.Type == PodReady`, but just to be safe i add a check here so it will fall back to default `rayv1.UnknownReason` non empty reason